R/checkDeps.R

```diff
@@ -23,22 +25,50 @@ checkDeps <- function(descriptionFile = ".", dependencyTypes = c("Depends", "Imp
 }
 
 checkRequirement <- function(package, version) {
-  requiredVersion <- package_version(if (version == "*") "0.0" else gsub("[^0-9.]", "", version))
-
-  packageVersion <- tryCatch({
-    utils::packageVersion(package)
-  }, error = function(error) NA)
-
+  # get installed package or R version
   if (package == "R") {
     packageVersion <- getRversion()
+  } else {
+    packageVersion <- tryCatch({
+      utils::packageVersion(package)
+    }, error = function(error) NA)
   }
 
+  # if the package is not installed
   if (is.na(packageVersion)) {
     warning(package, " is required, but not installed - please install it.")
-  } else if (packageVersion < requiredVersion) {
-    warning(package, " >= ", requiredVersion, " is required, but ", packageVersion,
-            " is installed - please update.")
-  }
+    result <- FALSE
+  # if we don"t care about the version
+  } else if ("*" == version) {
+    result <- TRUE
+  # compare version requirement
+  } else {
+    validops <- c("<", "<=", ">", ">=", "==", "!=")
+
+    # split version into operator and version number
+    opANDv <- unlist(
+      regmatches(x = version,
+                 m = gregexpr(pattern = paste0("(",
+                                               paste(c(validops, "[0-9\\.-]+"),
+                                                     collapse = "|"),
+                                               ")"),
+                              text = version))
+    )
 
-  return(!is.na(packageVersion) && packageVersion >= requiredVersion)
+    # compare to installed version
+    if (opANDv[1] %in% validops) {
+      result <- getFunction(name = opANDv[1])(packageVersion,
+                                            package_version(opANDv[2]))
+
+      # warn on mismatch
+      if (!result)
+        warning(paste(package, opANDv[1], package_version(opANDv[2]),
+                      "is required, but", packageVersion,
+                      "is installed - please update."))
```

Comment:
"please update" does not fit for e.g. `<=`, maybe something like:
```suggestion
                      "is installed - please install a compatible version."))
```